site/docs/concepts/architecture.md

```diff
@@ -2,27 +2,53 @@
+These may be existing open source or third party services, or custom ones. The module of a service indicates the capabilities it supports, the formats and interfaces, and how to deploy the service.  Modules may describe externally deployed services, or services deployed by fybrik.  Examples of modules are those that provide read/write access or produce implicit copies that serve as lower latency caches of remote assets.  
+Modules may also perform actions to enforce data governance policy decisions, such as masking or redaction as examples.
 
-As data assets may reside in different clusters/clouds a `Blueprint` CRD is created for each cluster, containing the information regarding the services to be deployed or configured in the given cluster. Depending on the setup the `PlotterController` will use various methods to distribute the blueprints. In a multi cluster setup the default distribution implementation is using [Razee](http://razee.io) to control remote blueprints, but several multi-cloud tools
-could be used as a replacement. The `PlotterController` also collects statuses and distributes
-updates of said blueprints. Once all the blueprints on all clusters are ready the plotter is marked as ready, and the overall status is propagated back to the user in the `FybrikApplication` status.
+## Connectors 
+Fybrik connects to external services to receive data governance decisions, metadata about datasets and credentials. Policies, assets and access credentials to the assets have to be defined before the user can run an application.
+
+The current abstraction supports 2 different [connectors](./connectors.md): one for data catalog (2) and one for policy manager (3). It is designed in an open way so that multiple different catalog and policy frameworks of all kinds of cloud and on-prem systems can be supported.
+
+The data steward configures policies  in an external policy manager over assets defined in an external data catalog. Dataset credentials are retrieved from Vault by using [Vault API](https://www.vaultproject.io/api) (6). Vault uses a custom secret engine implemented with [HashiCorp Vault plugins system](./vault_plugins.md) to retrieve the credentials from where they are stored (data catalog for example).
+
+## FybrikApplication
+Once the Modules and Connectors are installed, a developer can submit a `FybrikApplication` CRD to Kubernetes.
```

Comment:
Once Fybrik is deployed and configured, include catalogs, modules and connectors, a data user can submit ....